frontend/src/pages/HomePage.js

```diff
@@ -1,10 +1,36 @@
-import { Box, Button, Chip, Divider, Typography } from '@mui/material'
+import { Box, Chip, Divider, Typography } from '@mui/material'
 import FindMatch from '../components/FindMatch'
+import { findRoomSvc } from '../api/roomservice'
+import { UserContext } from '../contexts/UserContext'
+import { useContext, useEffect } from 'react'
 import { useNavigate } from 'react-router-dom'
 
 const HomePage = () => {
   const navigate = useNavigate()
 
+  const user = useContext(UserContext)
+
+  useEffect(() => {
+    const findExistingRoom = () => {
+      const filters = [{ id1: user.username }, { id2: user.username }]
+
+      filters.forEach(async (filter) => {
+        try {
+          const res = await findRoomSvc(filter)
```

Comment:
Correct me if I'm wrong, but we're trying to find rooms where either id1 or id2 is the user's username right? You can use a single query with an "or" operator instead of making 2 entire requests to the backend and then to mongoDB: https://www.mongodb.com/docs/manual/tutorial/query-documents/#specify-or-conditions.

Reply:
Yes will refractor all mongoDB related request in the next PR